fastcrypto-proving/benches/proving.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use ark_bls12_381::{Bls12_381, Fr as BlsFr};
+use ark_crypto_primitives::SNARK;
+use ark_ec::PairingEngine;
+use ark_ff::PrimeField;
+use ark_groth16::Groth16;
+use ark_relations::{
+    lc,
+    r1cs::{ConstraintSynthesizer, ConstraintSystemRef, SynthesisError},
+};
+use criterion::{
+    criterion_group, criterion_main, measurement::Measurement, BenchmarkGroup, BenchmarkId,
+    Criterion, SamplingMode,
+};
+
+#[derive(Copy, Clone)]
+struct DummyCircuit<F: PrimeField> {
+    pub a: Option<F>,
+    pub b: Option<F>,
+    pub num_variables: usize,
+    pub num_constraints: usize,
+}
+
+impl<F: PrimeField> ConstraintSynthesizer<F> for DummyCircuit<F> {
+    fn generate_constraints(self, cs: ConstraintSystemRef<F>) -> Result<(), SynthesisError> {
```

Comment:
I'd suggest at least a simple doc comment re the dummy circuit. TL;DR does it prove that "I know two numbers with whose product is X"? (in which case a, b are witnesses and X (their product) a public input)